Broken Python 3.5 without matplotlib

It seems that the flopy-3.3.1 release doesn't always work with Python 3.5, e.g.:
```
    import flopy
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/flopy/__init__.py:26: in <module>
    from . import modflow
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/flopy/modflow/__init__.py:1: in <module>
    from .mf import Modflow
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/flopy/modflow/mf.py:10: in <module>
    from ..mbase import BaseModel
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/flopy/mbase.py:22: in <module>
    from flopy import utils, discretization
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/flopy/utils/__init__.py:49: in <module>
    from .gridintersect import GridIntersect, ModflowGridIndices
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/flopy/utils/gridintersect.py:4: in <module>
    except ModuleNotFoundError:
E   NameError: name 'ModuleNotFoundError' is not defined
```
however this release works with Python 3.5 if matplotlib is installed, which is why the Travis CI tests run without error.

Two possible fixes:
 - Change `ModuleNotFoundError` to `ImportError` for Python 3.5 compatibility
 - Bump the minimum Python version to 3.6

--- a/flopy/export/utils.py
+++ b/flopy/export/utils.py
@@ -1508,7 +1508,7 @@ def export_contourf(filename, contours, fieldname='level', epsg=None,
 
     try:
         from shapely import geometry
-    except (ImportError, ModuleNotFoundError):
+    except ImportError:
         raise ImportError('export_contourf requires python shapely package')
 
     from ..utils.geometry import Polygon

--- a/flopy/utils/gridintersect.py
+++ b/flopy/utils/gridintersect.py
@@ -1,7 +1,7 @@
 import numpy as np
 try:
     import matplotlib.pyplot as plt
-except ModuleNotFoundError:
+except ImportError:
     plt = None
 
 from .geometry import transform
@@ -13,7 +13,7 @@ try:
     from shapely.affinity import translate, rotate
     from shapely.prepared import prep
     shply = True
-except ModuleNotFoundError:
+except ImportError:
     shply = False
 
 
@@ -107,7 +107,7 @@ class GridIntersect:
             msg = ("Shapely is needed for grid intersect operations! "
                    "Please install shapely if you need to use grid intersect "
                    "functionality.")
-            raise ModuleNotFoundError(msg)
+            raise ImportError(msg)
 
         self.mfgrid = mfgrid
         if method is None:
@@ -1240,16 +1240,16 @@ class GridIntersect:
         """
         try:
             from descartes import PolygonPatch
-        except ModuleNotFoundError:
+        except ImportError:
             msg = 'descartes package needed for plotting polygons'
             if plt is None:
                 msg = 'matplotlib and descartes packages needed for ' + \
                       'plotting polygons'
-            raise ModuleNotFoundError(msg)
+            raise ImportError(msg)
 
         if plt is None:
             msg = 'matplotlib package needed for plotting polygons'
-            raise ModuleNotFoundError(msg)
+            raise ImportError(msg)
 
         if ax is None:
             _, ax = plt.subplots()
@@ -1288,7 +1288,7 @@ class GridIntersect:
         """
         if plt is None:
             msg = 'matplotlib package needed for plotting polygons'
-            raise ModuleNotFoundError(msg)
+            raise ImportError(msg)
 
         if ax is None:
             _, ax = plt.subplots()
@@ -1333,7 +1333,7 @@ class GridIntersect:
         """
         if plt is None:
             msg = 'matplotlib package needed for plotting polygons'
-            raise ModuleNotFoundError(msg)
+            raise ImportError(msg)
 
         if ax is None:
             _, ax = plt.subplots()

